src/cffi.zig

```diff
@@ -42,7 +42,7 @@ const CModuleDefinitionInitOpts = extern struct {
     debug_name: ?[*:0]u8,
 };
 
-const CHostFunction = *const fn (userdata: ?*anyopaque, module: *core.ModuleInstance, params: [*]const Val, returns: [*]Val) void;
+const CHostFunction = *const fn (userdata: ?*anyopaque, module: *core.ModuleInstance, params: [*]const Val, returns: [*]Val) error{}!void;
```

Comment:
I think adding an empty error set here will make this function pointer incompatible with any C/FFI functions since it changes the signature. I made this test prototype and got:

```
src\main.zig:11:96: error: return type 'error{}!void' not allowed in function with calling convention 'x86_64_win'
export fn SomeFFIFunction(userdata: ?*anyopaque, params: [*]const Val, returns: [*]Val) error{}!void {
```

I think one way to resolve this is wrapping C functions in a trampoline function. The `ModuleImportsPackage` can have a special `addCHostFunction` or `addFFIHostFunction` that handles the wrapping and forwarding of the inner C function and userdata.

Reply:
I was wondering if that would be the case but since the build passed, I didn't think much of it. I should have tested this part. I'll work on getting this fixed.

Reply:
Thanks! And yeah ideally a failure would show up in the tests. I've been meaning to hook this up to a test (see #31), but work/other projects keep taking up my programming time.